core/xchaincore.go

```diff
@@ -311,7 +312,8 @@ func (xc *XChainCore) repostOfflineTx() {
 		}
 		xc.log.Debug("repost batch tx list", "size", len(batchTxMsg.Txs))
 		msgInfo, _ := proto.Marshal(batchTxMsg)
-		msg, _ := xuper_p2p.NewXuperMessage(xuper_p2p.XuperMsgVersion1, xc.bcname, header.GetLogid(), xuper_p2p.XuperMessage_BATCHPOSTTX, msgInfo, xuper_p2p.XuperMessage_SUCCESS)
+		got := snappy.Encode(nil, msgInfo)
```

Comment:
It's better to compress in `NewXuperMessage ` function so that the `p2p principle` can be migrated to other module easily.